browser/ui/webui/settings/brave_tor_snowflake_extension_handler.cc

```diff
@@ -0,0 +1,197 @@
+    default;
+
+void BraveTorSnowflakeExtensionHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "brave_tor.isSnowflakeExtensionAllowed",
+      base::BindRepeating(
+          &BraveTorSnowflakeExtensionHandler::IsSnowflakeExtensionAllowed,
+          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "brave_tor.isSnowflakeExtensionEnabled",
+      base::BindRepeating(
+          &BraveTorSnowflakeExtensionHandler::IsSnowflakeExtensionEnabled,
+          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "brave_tor.enableSnowflakeExtension",
+      base::BindRepeating(
+          &BraveTorSnowflakeExtensionHandler::EnableSnowflakeExtension,
+          base::Unretained(this)));
+
+  observation_.Observe(GetExtensionRegistry());
+}
+
+void BraveTorSnowflakeExtensionHandler::OnExtensionLoaded(
+    content::BrowserContext* browser_context,
+    const extensions::Extension* extension) {
+  DCHECK(extension);
+  if (extension->id() != kSnowflakeExtensionId ||
+      browser_context != web_ui()->GetWebContents()->GetBrowserContext()) {
+    return;
+  }
+  if (IsJavascriptAllowed()) {
+    FireWebUIListener("tor-snowflake-extension-enabled", base::Value(true));
+  }
+}
+
+void BraveTorSnowflakeExtensionHandler::OnExtensionUnloaded(
+    content::BrowserContext* browser_context,
+    const extensions::Extension* extension,
+    extensions::UnloadedExtensionReason reason) {
+  DCHECK(extension);
+  if (extension->id() != kSnowflakeExtensionId ||
+      browser_context != web_ui()->GetWebContents()->GetBrowserContext()) {
+    return;
+  }
+  if (IsJavascriptAllowed()) {
+    FireWebUIListener("tor-snowflake-extension-enabled", base::Value(false));
+  }
+}
+
+void BraveTorSnowflakeExtensionHandler::OnExtensionInstalled(
+    content::BrowserContext* browser_context,
+    const extensions::Extension* extension,
+    bool is_update) {
+  OnExtensionLoaded(browser_context, extension);
+}
+
+void BraveTorSnowflakeExtensionHandler::OnExtensionUninstalled(
+    content::BrowserContext* browser_context,
+    const extensions::Extension* extension,
+    extensions::UninstallReason reason) {
+  OnExtensionUnloaded(browser_context, extension,
+                      extensions::UnloadedExtensionReason::UNINSTALL);
+}
+
+extensions::ExtensionRegistry*
+BraveTorSnowflakeExtensionHandler::GetExtensionRegistry() {
+  return extensions::ExtensionRegistry::Get(
+      web_ui()->GetWebContents()->GetBrowserContext());
+}
+
+void BraveTorSnowflakeExtensionHandler::IsSnowflakeExtensionAllowed(
+    const base::Value::List& args) {
+  CHECK_EQ(args.size(), 1U);
+
+  Profile* profile = Profile::FromBrowserContext(
+      web_ui()->GetWebContents()->GetBrowserContext());
+
+  const bool is_allowed =
+      !profile->GetPrefs()->IsManagedPreference(tor::prefs::kTorDisabled) ||
+      profile->GetPrefs()->GetBoolean(tor::prefs::kTorDisabled);
```

Comment:
is this condition correct? shouldn't it be just `!kTorDisabled`?

Reply:
should be !managed || !disabled